zerver/actions/create_user.py

```diff
@@ -326,7 +331,8 @@ def process_new_human_user(
     # to keep all the onboarding code in zerver/lib/onboarding.py.
     from zerver.lib.onboarding import send_initial_direct_message
 
-    send_initial_direct_message(user_profile)
+    message_id = send_initial_direct_message(user_profile)
+    do_update_message_flags(user_profile, "add", "starred", [message_id])
```

Comment:
I'm surprised this added 3 database queries; we could think about just doing a `UserMessage.objects.filter()` query here, since we don't need all the `send_event` machinery.

Reply:
```diff
-    do_update_message_flags(user_profile, "add", "starred", [message_id])
+    UserMessage.objects.filter(user_profile=user_profile, message_id=message_id).update(
+        flags=F("flags").bitor(UserMessage.flags.starred)
+    )
```